scripts/pushJetpackComposeTags.main.kts

```diff
@@ -0,0 +1,140 @@
+
+import java.io.File
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+
+val libs = listOf(
+    "https://developer.android.com/jetpack/androidx/releases/compose-ui",
+    "https://developer.android.com/jetpack/androidx/releases/compose-material3",
+    "https://developer.android.com/jetpack/androidx/releases/collection",
+    "https://developer.android.com/jetpack/androidx/releases/annotation",
+    "https://developer.android.com/jetpack/androidx/releases/lifecycle",
+    "https://developer.android.com/jetpack/androidx/releases/navigation",
+)
+
+fun tagName(libName: String, version: String) = "androidx/$libName/$version"
+
+"git remote add aosp https://android.googlesource.com/platform/frameworks/support".execCommand().apply {
+    if (this.isNullOrEmpty()) {
+        println("Added aosp https://android.googlesource.com/platform/frameworks/support repo")
+    } else {
+        check(contains("already exists"))
+    }
+}
+"git fetch aosp".runCommand()
+
+for (lib in libs) {
+    println("=== Loading $lib")
+
+    fun Element.extractNameCommitVersion(): Triple<String, String, String> {
+        // <code translate="no" dir="ltr">androidx.lifecycle:lifecycle-viewmodel-savedstate:1.0.0-rc03</code>
+        // <code translate="no" dir="ltr">androidx.compose.ui:ui-*:1.2.0-alpha08</code>
+        val text = toString()
+        // androidx.compose.ui:ui-*:1.2.0-alpha08
+        val coords = text
+            .substringAfter("<code")
+            .substringAfter(">")
+            .substringBefore("</code>")
+
+        val group = coords.substringBefore(":") // androidx.compose.ui
+        val module = coords.substringAfter(":").substringBefore(":") // ui-*
+
+        val shortModule = module.replace("-*", "").replace(".", "-") // ui
+        val name = if (group.count { it == '.' } > 1) {
+            val shortGroup = group.substringAfter(".").substringBeforeLast(".").replace(".", "-") // compose
+            "$shortGroup-$shortModule"
+        } else {
+            shortModule
+        }
+        val commit = text.substringAfter("frameworks/support/+log/").run {
+            if (contains("..")) {
+                substringAfter("..")
+            } else {
+                this
+            }
+        }.substringBefore("/").substringBefore("\"").validateCommit()
+        val version = text.substringAfter("Version ").substringBefore(" contains").validateVersion()
+        return Triple(name, commit, version)
+    }
+
+    Jsoup
+        .connect(lib)
+        .get()
+        .select("*")
+        .asSequence()
+        .filter { it.tagName() == "p" && it.toString().contains("these commits") }
+        .map { it.extractNameCommitVersion() }
+        .forEach { (name, commit, version) ->
+            val tagName = tagName(name, version)
+            println("Creating tag $tagName on $commit")
+            "git tag -f $tagName $commit".runCommand()
```

Comment:
Improvement: Should we create only not existant one?
I'd suggest requirest current tags (one command with list will be faster than checking one-by one) and filter that list by it

Reply:
Done